### PR TITLE
CFE-2886: Fixing the last non-performance warnings from LGTM

### PR DESCRIPTION
**Pull request by olehermanse**

### CODE REVIEW

**Comment by olehermanse** [comment]
Replaces #3245

File: cf-monitord/env_monitor.c [hunk 1]
```diff
@@ -106,7 +106,7 @@ static void LeapDetection(void);
 static Averages *GetCurrentAverages(char *timekey);
 static void UpdateAverages(EvalContext *ctx, char *timekey, Averages newvals);
 static void UpdateDistributions(EvalContext *ctx, char *timekey, Averages *av);
-static double WAverage(double newvals, double oldvals, double age);
+static double WAverage(double new_val, double old_val, double age);
 static double SetClasses(EvalContext *ctx, char *name, double variable, double av_expect, double av_var, double localav_expect,
```
Comment (karlhto) [inline cf-monitord/env_monitor.c]: Newline earlier here, maybe

File: cf-monitord/env_monitor.c [hunk 2]
```diff
@@ -849,71 +849,75 @@ static void UpdateDistributions(EvalContext *ctx, char *timekey, Averages *av)
 
 /*****************************************************************************/
 
+/*
+    This function performs a weighted average of an old and a new measured
+    value. Weights depend on the age of the data. If one or both values
+    are "unreasonably" large (>9999999) they will be ignored.
+*/
 /* For a couple of weeks, learn eagerly. Otherwise variances will
```
Comment (karlhto) [inline cf-monitord/env_monitor.c]: Why not merge these comment blocks?

**Comment by karlhto** [comment]
Or not. 
```
cf3lex.l: In function ‘yyuseraction’:
cf3lex.l:70:1: error: ‘P’ undeclared (first use in this function)
 P.offsets.current += yyleng;
 ^
cf3lex.l:70:1: note: each undeclared identifier is reported only once for each function it appears in
cf3lex.l: In function ‘yylex’:
cf3lex.l:139:32: error: ‘P’ undeclared (first use in this function)
                           free(P.current_line);
                                ^
make[3]: *** [cf3lex.lo] Error 1
make[3]: *** Waiting for unfinished jobs....
cf3parse.y: In function ‘ParseErrorVColumnOffset’:
cf3parse.y:1248:46: error: ‘P’ undeclared (first use in this function)
     fprintf(stderr, "%s:%d:%d: error: %s\n", P.filename, P.line_no, P.line_pos + column_offset, errmsg);
                                              ^
cf3parse.y:1248:46: note: each undeclared identifier is reported only once for each function it appears in
cf3parse.y: In function ‘ParseWarningV’:
cf3parse.y:1292:11: error: ‘P’ undeclared (first use in this function)
     if (((P.warnings | P.warnings_error) & warning) == 0)
           ^
cf3parse.y: In function ‘fatal_yyerror’:
cf3parse.y:1342:90: error: ‘P’ undeclared (first use in this function)
     fprintf(stderr, "%s: %d,%d: Fatal error during parsing: %s, near token \'%.20s\'\n", P.filename, P.line_no, P.line_pos, s, sp ? sp : "NULL");
                                                                                          ^
cf3parse.y: In function ‘yyparse’:
cf3parse.y:103:57: error: ‘P’ undeclared (first use in this function)
                            ParserDebug("P:bundle:%s\n", P.blocktype);
                                                         ^
cf3parse.y:68:51: note: in definition of macro ‘ParserDebug’
 #define ParserDebug(...) LogDebug(LOG_MOD_PARSER, __VA_ARGS__)
```

**Comment by olehermanse** [comment]
@karlhto see labels ;)

File: cf-monitord/env_monitor.c (hunk 1 above)
Comment (olehermanse) [reply cf-monitord/env_monitor.c]: Not part of my changes ;)

File: cf-monitord/env_monitor.c (hunk 2 above)
Comment (olehermanse) [reply cf-monitord/env_monitor.c]: It's more clear who wrote what.

**Comment by karlhto** [comment]
I just saw "requested for review" and went for it.

**Comment by olehermanse** [comment]
This pull request **fixes 5 alerts** when merging e406321f49283e7ee50cf620eb3c88207b460889 into 77acf919c9cb8c479e5e7c13b6b657892e84cf49 - [view on LGTM.com](https://lgtm.com/projects/g/cfengine/core/rev/pr-e652555870d300b9de6073c13c36e5365e62376b)

**fixed alerts:**

* 3 for Short global name
* 1 for Bad check for oddness
* 1 for Comparison result is always the same

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by olehermanse** [comment]
@cf-bottom please build this in jenkins with exotics.

**Comment by cf-bottom** [comment]
Alright, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/1097/)

(with exotics)

https://ci.cfengine.com/job/pr-pipeline/1097/

**Comment by olehermanse** [comment]
This pull request **fixes 5 alerts** when merging e406321f49283e7ee50cf620eb3c88207b460889 into 11b10fb5d80a67763977c8f2dac4dc850156aa4c - [view on LGTM.com](https://lgtm.com/projects/g/cfengine/core/rev/pr-11d29382640b67b3e88265f6075f781b5ccf5be2)

**fixed alerts:**

* 3 for Short global name
* 1 for Bad check for oddness
* 1 for Comparison result is always the same

---

*Comment posted by [LGTM.com](https://lgtm.com)*

File: cf-monitord/env_monitor.c [hunk 3]
```diff
@@ -849,71 +849,75 @@ static void UpdateDistributions(EvalContext *ctx, char *timekey, Averages *av)
...
    be way too large. Then downplay newer data somewhat, and rely on
    experience of a couple of months of data ... */
 
-static double WAverage(double anew, double aold, double age)
+static double WAverage(double new_val, double old_val, double age)
 {
-    double av, cf_sane_monitor_limit = 9999999.0;
-    double wnew, wold;
-
```
Comment (craigcomstock) [inline cf-monitord/env_monitor.c]: what? wold isn't clear? ;) wow, woot, wren(ame)

**Pull request merged**